Identity/Crypto.cs

```diff
@@ -40,6 +44,16 @@ internal static byte[] ObfuscateUserId(byte[] userId, byte[] trustchainId)
                 return GenericHash(toHash, BlockHashSize);
             }
 
+            internal static byte[] GenerateAppID(byte[] appSecret)
+            {
+                byte[] publicKey = new byte[PublicKeySize];
+                Array.Copy(appSecret, AppSecretSize - PublicKeySize, publicKey, 0, PublicKeySize);
+                byte[] start = new byte[1 + AuthoreSize];
+                start[0] = AppCreationNature;
+                var toHash = ConcatByteArrays(start, publicKey);
```

Comment:
There is `AuthorSize` byte of emptyness here ? Is it what you want ?

Reply:
yep, I need zeros here. Which is what c# does apparently